packages/atclient/include/atclient/monitor.h

```diff
@@ -246,6 +246,11 @@ int atclient_monitor_pkam_authenticate(atclient *monitor_conn, const char *atser
  */
 void atclient_monitor_set_read_timeout(atclient *monitor_conn, const int timeout_ms);
 
+typedef struct atclient_monitor_hooks {
+  int (*pre_decrypt_notification)(void);
+  int (*post_decrypt_notification)(void);
+} atclient_monitor_hooks;
+
```

Comment:
Monitor hooks, literally just allow you to run some code in between steps in the monitor read function, unlike the connection, they don't expose any internal values.